### ISSUE TITLE
FVS: Add GRM calibration data to FVS IN

### ISSUE THREAD

**Issue by lbross** [USFS-PNW/Fia-Biosum-Manager#166]

**Comment by jsfried** [comment]
I built a new project for Plumas County California based on the latest FIADB (with 7 years of remeasure data suitable for GRM). I learned that one does not need to specify a GRM EvalID to use the GRM option-- remeasure data appeared in the FVS input files for remeasured trees even using a current area current volume evalid. GRM data loads into the db\master_aux database and includes all trees with GRM data. However, only trees classified as "SURVIVOR" have their GRM data propagate through to the FVSIn.FVS_TreeInit table. We might consider loading only the SURVIVOR records into MASTER_AUX.GRM_TREE in the future to reduce confusion and database size (the other records will likely never be used). 

I checked that the correct previous diameter and height values propagate to the FVSin (they do). I ran the FVSIn database through FVS and studied the FVS documentation on how this data is used. By default, GRM tree data is only used to guide projection of trees of a given species in a given stand when there are 5 or more trees of that species in the stand. So if your stand had 5 ponderosa, 2 sugar, 3 douglas fir and 4 incense cedar, only the ponderosas would get their basal area (diameter) growth adjusted (calibrated) from the GRM data. Moreover, the previous height data is used only for adjusting the diameter (yes, not a typo) growth of trees <5" DBH. One can change the threshold of the number of trees of a species that must be present before calibration occurs for that species-- I did, changing it to 3, and the result was many more trees getting a different projected diameter, as compared to the threshold of 5. The prescription I ran was a thin from below, and I looked only at trees that were present through the entire 3 decades of simulation (comparing their size at year 31 with no calibration, calibration if 5 trees and calibration if 3 trees). There were (tiny) differences in the sizes of even trees that did not meet the thresholds presumably due to changes in the competitive relationships among trees resulting from the adjustment of tree size in the species that did meet the thresholds, so there is a lot going on. Also, FVS reduces the calibration impact over time, such that by year 30, most of the effect of calibration is attenuated (assymtotes back towards the original, uncalibrated values), so sleuthing through the data to understand the total effect of calibration will take much more time than I have right now. The good news is that the GRM data does seem to get loaded from FIADB and passed on the FVS (where it is up to the analyst to make valid use of it through thoughtful selection of FVS keywords and parameters, and testing, testing, testing). The only thing to consider in a future release is limiting the GRM data in the MASTER_AUX to just those records with micr_component_al_forest="SURVIVOR". One note of caution-- the GRM_STAND table in this project contains many more stands than appear in the GRM_TREE table, in part because the former includes plots with INVYR 2011 (which would have been remeasure plots), but the GRM_TREE table does not (nor does the TREE_GRM_COMPONENT table in FIADB from which this data are loaded). I am inquiring as to why this is the case, but I don't think the absence of this tree GRM data causes any inconsistencies for BioSum in terms of how FVSIn files are made. @sorgtyler @lbross
